[Distributed] Adding Python tests for the TCPStore getNumKeys and deleteKey

osalpekar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45223 [Distributed] Adding Python tests for the TCPStore getNumKeys and deleteKey**
* #43963 [Distributed] DeleteKey API for c10d TCP Store
* #43962 [Distributed] getNumKeys API to c10d TCPStore

Previous diffs in this stack implemented the getNumKeys and deleteKey
APIs in the c10d Store as well as added tests at the C++ layer. This diff adds
tests at the Python level in test_c10d.py

Differential Revision: [D23878455](https://our.internmc.facebook.com/intern/diff/D23878455/)

osalpekar committed:
[Distributed] Adding Python tests for the TCPStore getNumKeys and deleteKey

Previous diffs in this stack implemented the getNumKeys and deleteKey
APIs in the c10d Store as well as added tests at the C++ layer. This diff adds
tests at the Python level in test_c10d.py

Differential Revision: [D23878455](https://our.internmc.facebook.com/intern/diff/D23878455/)

[ghstack-poisoned]

pritamdamania87 commented on test/distributed/test_c10d.py:
```diff
@@ -272,6 +272,26 @@ def test_address_already_in_use(self):
+        fs.add("key3", 1)
+        fs.set("key1", "value1")
+        self.assertEqual(fs.num_keys(), 5)
+        fs.delete_key("key")
```
Shouldn't we validate the value of "key" using get?

pritamdamania87 commented on test/distributed/test_c10d.py:
```diff
@@ -272,6 +272,26 @@ def test_address_already_in_use(self):
+        self.assertEqual(fs.num_keys(), 5)
+        fs.delete_key("key")
+        self.assertEqual(fs.num_keys(), 4)
+        fs.delete_key("key0")
+        fs.delete_key("key3")
+        self.assertEqual(fs.num_keys(), 2)
+        fs.set("key4", "value2")
+        self.assertEqual(fs.num_keys(), 3)
+
```
Add tests which try to retrieve the deleted keys using get.

osalpekar committed:
Update on "[Distributed] Adding Python tests for the TCPStore getNumKeys and deleteKey"


Previous diffs in this stack implemented the getNumKeys and deleteKey
APIs in the c10d Store as well as added tests at the C++ layer. This diff adds
tests at the Python level in test_c10d.py

Differential Revision: [D23878455](https://our.internmc.facebook.com/intern/diff/D23878455/)

[ghstack-poisoned]

osalpekar committed:
Update on "[Distributed] Adding Python tests for the TCPStore getNumKeys and deleteKey"


Previous diffs in this stack implemented the getNumKeys and deleteKey
APIs in the c10d Store as well as added tests at the C++ layer. This diff adds
tests at the Python level in test_c10d.py

Differential Revision: [D23878455](https://our.internmc.facebook.com/intern/diff/D23878455/)

[ghstack-poisoned]

osalpekar committed:
Update on "[Distributed] Adding Python tests for the TCPStore getNumKeys and deleteKey"


Previous diffs in this stack implemented the getNumKeys and deleteKey
APIs in the c10d Store as well as added tests at the C++ layer. This diff adds
tests at the Python level in test_c10d.py

Differential Revision: [D23878455](https://our.internmc.facebook.com/intern/diff/D23878455/)

[ghstack-poisoned]
